src/Admin/BaseBlockAdmin.php

```diff
@@ -147,7 +175,14 @@ public function postPersist($object)
      */
     public function preRemove($object)
     {
-        $this->blockManager->get($object)->preRemove($object);
+        if (\is_callable([$object, 'preRemove'])) {
+            $this->blockManager->get($object)->preRemove($object);
+
+            @trigger_error(
+                'The '.__METHOD__.' method is deprecated since version 3.x and will be removed in sonata/block-bundle 4.0.',
```

Comment:
```suggestion
                'The '.__METHOD__.'() method is deprecated since sonata-project/block-bundle 3.x and will be removed in version 4.0.',
```